puma_motor_driver/include/puma_motor_driver/driver.h

```diff
@@ -89,19 +88,20 @@ class Driver
   void setEncoderCPR(uint16_t encoder_cpr);
   void setGearRatio(float gear_ratio);
   void setMode(uint8_t mode);
-  void setMode(uint8_t mode, float p, float i, float d);
-  void setGains(float p, float i, float d);
+  void setMode(uint8_t mode, double p, double i, double d);
+  void setGains(double p, double i, double d);
```

Comment:
IIRC, the underlying firmware doesn't even allow you to change the gains when the controller is active (ie, in motion), for good reason. This API should perhaps reflect that? Hmm.

Either way, I don't think it's a major hardship to have to respecify when you are setting the constants what mode they are for.